src/core/js/views/notifyView.js

```diff
@@ -191,33 +197,44 @@ define([
         },
 
         closeNotify: function (event) {
-            //prevent from being invoked multiple times - see https://github.com/adaptlearning/adapt_framework/issues/1659
+            // Prevent from being invoked multiple times - see https://github.com/adaptlearning/adapt_framework/issues/1659
             if (!this.isOpen) return;
             this.isOpen = false;
 
-            if (this.disableAnimation) {
+            var complete = function() {
+                if (this.disableAnimation) {
 
-                this.$('.notify-popup').css('visibility', 'hidden');
-                this.$el.css('visibility', 'hidden');
+                    this.$('.notify-popup').css('visibility', 'hidden');
+                    this.$el.css('visibility', 'hidden');
 
-                this.remove();
+                    this.remove();
 
-            } else {
+                } else {
 
-                this.$('.notify-popup').velocity({ opacity: 0 }, {duration: 400, complete: function() {
-                    this.$('.notify-popup').css('visibility', 'hidden');
-                }.bind(this)});
+                    this.$('.notify-popup').velocity({ opacity: 0 }, {duration: 400, complete: function() {
+                        this.$('.notify-popup').css('visibility', 'hidden');
+                    }.bind(this)});
 
-                this.$('.notify-shadow').velocity({ opacity: 0 }, {duration: 400, complete:function() {
-                    this.$el.css('visibility', 'hidden');
-                    this.remove();
-                }.bind(this)});
-            }
+                    this.$('.notify-shadow').velocity({ opacity: 0 }, {duration: 400, complete:function() {
+                        this.$el.css('visibility', 'hidden');
+                        this.remove();
+                    }.bind(this)});
+                }
+
+                $('body').scrollEnable();
+                $('html').removeClass('notify');
 
-            $('body').scrollEnable();
-            $('html').removeClass('notify');
+                // Return focus to previous active element
+                Adapt.trigger('popup:closed notify:closed', this.$previousActiveElement);
+            }.bind(this);
 
-            Adapt.trigger('popup:closed notify:closed');
+            // If closeNotify is called before showNotify has finished then wait
+            // until it's open.
+            if (!this.hasOpened) {
```

Comment:
Are we not able to avoid the `complete()` function and return early at the top of this function with a debounce on `showNotify()`? Or is there an issue with a debounced call to itself?